### ISSUE TITLE
(3P) External Media: Arrow keys not captured by modal

### ISSUE THREAD

**Issue by jeryj** [Automattic/jetpack#15855]
When using a Media Extensions modal (like Pexels or Google photos), you can press the arrow keys and it will either close the modal or interact with the editor canvas.

#### Steps to reproduce the issue

**Modal closing**
1. Insert an image block
2. Click on Select Image
3. Choose "Pexels Free Libary"
4. See the modal open
5. Press an arrow key
6. The modal will close

**Interacting with window behind the modal**
1. Insert an image block
2. Click on Select Image
3. Choose "Google Libary"
4. See the modal open
5. Add a Before or After date filter
6. Click the Before/After filter button to get the calendar to show
7. press the arrow keys
8. Focus is now in the editor canvas. You can tab/arrow key around and see things change behind the modal.

#### What I expected

Arrow keys should be bound within the modal.

**Comment by obenland** [comment]
Possible ways:
* Check the modal if there are ways to better implement ours as these keypresses should be handled by it.
* Check onClose handlers to see if they're too broad

**Comment by WunderBart** [comment]
I did some debugging and the modal seems to get closed when the `setSelectedSource` state update is called within `onClose` callback. When I comment [this line out](https://github.com/Automattic/jetpack/blob/644249810bbf9d1cf0336a19116445b6ad97d384/extensions/shared/external-media/media-button/index.js#L32) it behaves like other modals (which means arrows not doing anything), like i.e. the [Keyboard Shortcuts Help Modal](https://github.com/WordPress/gutenberg/blob/dbcaf38e886781d0bf5101db8cd04b5360ce2685/packages/edit-post/src/components/keyboard-shortcut-help-modal/index.js):
<p align="center"><img width="240" src="https://d.pr/i/OQIqDh+"></p>

So our fix could be possibly making that state update **not** close the modal?

**Comment by WunderBart** [comment]
Fixed by https://github.com/Automattic/jetpack/pull/16055
